src/queryService.ts

```diff
@@ -0,0 +1,28 @@
+import { graphql } from '@octokit/graphql';
+import { AuthenticationSession } from 'vscode';
+
+const getRecentPullRequests = async (repositoryName: String, owner: String, limit: Number, session: AuthenticationSession) => {
+    const res = await graphql({
+		query: `query pullRequests($name: String!, $owner: String!, $limit: Int!) {
+			repository(name: $name, owner: $owner) {
+			  pullRequests(last: $limit) {
+				nodes {
+				  changedFiles
+				  createdAt
+				  body
+				  author {
+					login
+				  }
+				}
+			  }
+			}
+		  }`,
+		owner, // TODO get these values from extension api
+		name: repositoryName,
+        limit,
+		headers: { authorization: `Bearer ${session.accessToken}` },
```

Comment:
In theory, this query service could get the session instead of having it being done in `extension.ts` and be passed in to `getRecentPullRequests` .. something to play with.

Reply:
Since we're in hackathon mode, I wouldn't worry about this too much but I think it's worth thinking about how a user will "log in" to the extension when they first install it. If a user is not logged in, the dialog box pops up and says "HEY YOU NEED TO LOG IN", that's not the best experience. Maybe a notification message (`showInformationMessage` API) would be softer at first. @eamodio did you have any ideas for that?